torch/onnx/_internal/fx/passes/_utils.py

```diff
@@ -28,6 +28,40 @@ def wrapped(*args):
     return wrapped
 
 
+@_beartype.beartype
+def set_node_name(
+    node: torch.fx.Node,
+    new_name: str,
+    name_to_node_cache: Optional[Dict[str, torch.fx.Node]] = None,
+):
+    """Safely set the unique name of a node.
+
+    If the new name is already taken by another node, the name of the other node will be
+    updated to be "{new_name}.1". This function will recursively update the names until
+    there is no conflict.
+
+    To avoid recomputing the name_to_node_cache, it can be provided as an argument. If
+    provided, the caller is responsible for ensuring the cache is accurate and in sync
+    with the owning module of the node.
+
+    Args:
+        node: The node to update.
+        new_name: The new name to use.
+        name_to_node_cache: A cache of node names to nodes. If not provided, this
+            function will build the cache from the owning module of the node.
+    """
+    module = node.graph.owning_module
+    name_to_node_cache = name_to_node_cache or {
+        _node.name: _node for _node in module.graph.nodes
+    }
+
+    if new_name in name_to_node_cache and name_to_node_cache[new_name] != node:
+        set_node_name(name_to_node_cache[new_name], f"{new_name}.1")
```

Comment:
Avoid recursion when processing nodes. There is no guarantee how many `assignments` can happen in a graph. On the other hand, can we create names such as `x.1`, `x.2`, etc? Names such as `x.1` and `x.1.1.1` are hard to read.